ste/init.go

```diff
@@ -162,6 +162,7 @@ func ExecuteNewCopyJobPartOrder(order common.CopyJobPartOrderRequest) common.Cop
 	jppfn := JobsAdmin.NewJobPartPlanFileName(order.JobID, order.PartNum)
 	jppfn.Create(order)                                                                   // Convert the order to a plan file
 	jpm := JobsAdmin.JobMgrEnsureExists(order.JobID, order.LogLevel, order.CommandString) // Get a this job part's job manager (create it if it doesn't exist)
+	jpm.CredentialType(order.CredentialInfo.CredentialType)
```

Comment:
Hang the phone-- Something smells severely wrong here.

Initially, I was going to point you to InMemoryTransitJobState for obtaining the information you're trying to carry here. However, I realized upon thinking about that that, the credentials we pass along to the STE are **_always_** for the destination.

S3 is never a destination; therefore it won't get passed along properly here. Thus, I don't think your check later in the STE ever truly works. Did you test your code? Secondarily, if you did, could you try attaching a debugger to your addition to the s3 Source Info Provider and see if your check ever goes off?

My theory is that we may not even need to bother with this check in the STE.

Reply:
I have tested this and it does pass the credentials and the check in S3 SIP does go off. Without passing the source Credential Type, the jptm does not know the correct URL to retrieve contents from.

Reply:
Kind of agree with Adele here. Having source credentials in jptm seems wierd. We need the cred type to sign S3 source we construct PBTC. This makes having keyword for AWS_ACCESS_KEY_ID a bit cleaner solution.